CodeStarterCamp_Week3/step2.swift

```diff
@@ -35,27 +39,30 @@ class CoffeeShop {
     var sales: Int
     var barista: Person
     var menu: [Coffee: Int]
-    var pickUpTable: [String]
     
-    init(sales: Int, barista: Person, menu: [Coffee: Int], pickUpTable: [String]) {
+    init(sales: Int, barista: Person, menu: [Coffee: Int]) {
         self.sales = sales
         self.barista = barista
         self.menu = menu
-        self.pickUpTable = pickUpTable
-    }
-    
-    func takeOrder(coffee: Coffee) {
-        print("\(coffee.rawValue)의 주문이 들어왔습니다.")
     }
     
     func make(_ coffee: Coffee, from name: String) {
-        pickUpTable.append(coffee.rawValue)
-        if let coffeeName = pickUpTable.last {
-            print("\(name) 님이 주문하신 \(coffeeName)(이/가) 준비되었습니다. 픽업대에서 가져가주세요.")
+        if let price = menu[coffee] {
+            sales += price
+        }
+        
+        var pickUpTable: [Coffee] = [] {
+            didSet {
+                if let coffeeName = pickUpTable.last {
+                    print("\(name)님이 주문하신 \(coffeeName.rawValue)(이/가) 준비되었습니다. 픽업대에서 가져가주세요.")
+                }
+            }
```

Comment:
didSet을 잘 활용해주셨네요!
하지만 pickUpTable을 CoffeeShop이 프로퍼티로 갖게 하라는 요구사항이 지켜지지 않게 되었어요. 🥲

프로퍼티로 선언했을 때 손님의 이름을 받기 어렵다고 느끼셨다면 pickUpTable의 타입을 아래와 같이 튜플을 활용해 전달해줄 수 있습니다.💭
또는 방금 주문한 손님의 이름을 프로퍼티로 추가해서 해결할 수도 있겠네요. 🤔

```suggestion
var pickUpTable: [(menu: Coffee, name: String)] = [] {
    didSet {
        let order = pickUpTable.removeFirst()
        print("\(order.name)님이 주문하신 \(order.menu.rawValue)(이/가) 준비되었습니다. 픽업대에서 가져가주세요.")
    }
}
```